Exclude API docs from basic authentication requirement

yotommy opened the pull request:
### Description

Exclude API docs from basic authentication requirement.

## Release notes

Exclude API docs from basic authentication requirement. They are meant to be [publicly visible](https://sourcegraph.com/github.com/civiform/civiform@9358f1b66ff84c74f953246520cf86544568dcb9/-/blob/server/app/views/api/ApiDocsView.java?L280-288).

### Checklist

#### General

Read the full guidelines for PRs [here](https://docs.civiform.us/contributor-guide/developer-guide/technical-contribution-guide#guidelines-for-pull-requests)

- [x] Added the correct label: < feature | bug | dependencies | infrastructure | ignore-for-release | database >
- [x] Created unit and/or browser tests which fail without the change (if possible)
- [x] Performed manual testing (Chrome and Firefox if it includes front-end changes)
- [ ] Extended the README / documentation, if necessary

### Issue(s) this completes

Fixes #5353

bion reviewed (commented):
@yotommy possible to test this with a browser tes?

avaleske commented:
+1 on browser tests, everything else is lgtm

yotommy commented:
Added a browser test, which was an outstanding idea, as it exposed a bug: only admins could change programs on the API docs. Added `api-docs.ts` to the `applicant-bundle` to fix this.

The browser test encountered authentication errors (returning 401 even though it was using `AnonymousClient`), so I modified the existing regex to exclude the API docs path rather than defining a new pattern/client stanza.

avaleske commented on server/app/assets/javascripts/applicant_entry_point.ts:
```diff
@@ -25,4 +26,5 @@ window.addEventListener('load', () => {
   azureDelete.init()
   azureUpload.init()
   phoneNumber.init()
+  apiDocs.init()
```
suggestion (not blocking): would it be helpful to add a comment here explaining why the api docs are in the applicant bundle?

avaleske reviewed (approved):
Thank you!

bion reviewed (approved):
nice!
